src/blueprints/settings.py

```
@@ -1123,13 +1262,33 @@ def client_log():
+_shutdown_lock = threading.Lock()
+
+
 @settings_bp.route("/shutdown", methods=["POST"])
 def shutdown():
     """Reboot or shut down the device.
 
-    Consider adding authentication or CSRF protection if publicly exposed.
+    Rate-limited to one call per 30 seconds to prevent accidental repeats.
     """
-    data = request.get_json() or {}
+    global _last_shutdown_time
```

Ruff (0.15.7)

[warning] Using the global statement to update `_last_shutdown_time` is discouraged

(PLW0603)